urlpattern/resources/urlpatterntestdata.json

```diff
@@ -1202,10 +1202,11 @@
   {
     "pattern": [{ "protocol": "http", "port": "80 " }],
     "inputs": [{ "protocol": "http", "port": "80" }],
-    "exactly_empty_components": ["port"],
-    "expected_match": {
-      "protocol": { "input": "http", "groups": {} }
-    }
+    "expected_obj": {
+      "protocol": "http",
+      "port": "80"
```

Comment:
So this mean, the new changes remove trailing leading spaces on canonicalize_port step right?